packages/loader/container-loader/src/container.ts

```diff
@@ -96,6 +96,8 @@ import { PrefetchDocumentStorageService } from "./prefetchDocumentStorageService
 import { parseUrl, convertProtocolAndAppSummaryToSnapshotTree } from "./utils";
 
 const detachedContainerRefSeqNumber = 0;
+// TODO: move and expose this
+const shouldRefresh = true;
```

Comment:
How should loader/container get this information from host?

Alternatively it could be a runtime decision, but I'm not sure about that. i.e. Container does not know difference, but ContainerRuntime never returns if it chooses raceToSummarize approach, and instead raises event indicating page refresh.

Reply:
@anthony-murphy @vladsud - interested in thoughts here specifically

Reply:
i added some comments below, with my latest change you should do it after we check if the context reload is needed i think. in that case only clients that need to close with race. other clients will just behave normally 

---
In reply to: [514456907](https://github.com/microsoft/FluidFramework/pull/4096#discussion_r514456907) [](ancestors = 514456907)

Reply:
Always races to summarize